### DISCUSSION TITLE
next/head is not working with next13?

### DISCUSSION THREAD

**Discussion by thmohd** [vercel/next.js#42161] [Help] [answered]
upgrading to nextj13 with app directory , however i noticed that <Head> is no longer showing data per page? is this a bug or am i missing something?

**Comment by dejurin** [comment, marked as answer]
In the app directory, you can modify the <head> tag using a new special head.js file. This replaces [next/head](https://nextjs.org/docs/api-reference/next/head) in the pages directory.

https://beta.nextjs.org/docs/routing/pages-and-layouts#modifying-head


![](https://assets.vercel.com/image/upload/f_auto,q_100,w_1600/v1666704760/nextjs-docs/darkmode/head-file.png)

**Comment by tmb-github** [comment]
I've discovered the file does not have to be located in the same directory. I've made a parameterized Head component (as described in the 13 beta documentation) that I've placed in my components directory, and I simply import it into the page.js as Head, and it works fine. This eliminates the need to have a separate head.js in each app folder.

I'm also using a static set of meta tags that are defined in the main layout.js file; dynamic head elements--ones that need to change from page-to-page--are created with the parameterized Head component imported into the individual pages.

The current documentation on this (https://beta.nextjs.org/docs/api-reference/file-conventions/head) seems to be in error, though:

`Warning: Currently, the Head export does not re-render on client-side transitions, only on initial render. To work around this for <title>, you can use a client component with useEffect that updates document.title. This will be fixed soon in a future release.`

The Head component called on each page does cause the previously set dynamic tags to be replaced.

**Reply by thmohd** [reply]
definitely i will try that, bcz it feels odd to me to fetch data in the head.js again

**Comment by tmb-github** [comment]
@thmohd Below is my design pattern.

IMPORTANT: NextJS puts the dynamically generated head elements at the BOTTOM of the `<head>` element, so look for changes there, otherwise it may seem that it's not updating the `<head>`. The static head elements remain untouched through SPA navigation.

app/layout.js:

```
export default function RootLayout({ children }) {
  return (
    <html lang="en-us">
      <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta name="apple-mobile-web-app-capable" content="yes" />
        [...and so on with other fixed tags that appear on all pages...]
      </head>
      <body>
        <Header />
        {children}
        <Footer />
      </body>
    </html>
  );
}
```
components/MicroHead/index.js:
```
export default async function Head({ canonical, metaDescription, title }) {
  return (
    <>
      <title key="title">{title}</title>
      <meta name="description" content={metaDescription} key="description" />
      <meta
        name="twitter:description"
        content={metaDescription}
        key="twitter:description"
      />
      <meta name="twitter:title" content={title} key="twitter:title" />
      <meta name="twitter:url" content={canonical} key="twitter:url" />
      <meta
        property="og:description"
        content={metaDescription}
        key="og:description"
      />
      <meta property="og:title" content={title} key="og:title" />
      <meta property="og:url" content={canonical} key="og:url" />
    </>
  );
}
```
app/page.js:
```
import Head from "../components/MicroHead";

export default function Home() {
  return (
    <>
      <Head
        canonical="https://localhost/shell/"
        title="Home | Shell"
        metaDescription="Home page description for SHELL application [70 characters are best here]."
      ></Head>
      <main>
        <h1>Home</h1>
        <p>Home page of Shell application</p>
      </main>
    </>
  );
}
```
app/dummy1/page.js:
```
import Head from "../../components/MicroHead";

export default function Dummy1() {
  return (
    <>
      <Head
        canonical="https://localhost/shell/dummy2"
        title="Dummy 1 | Shell"
        metaDescription="Dummy 1 page description for SHELL application [70 characters are best here]."
      ></Head>

      <main>
        <h1>Dummy 1</h1>
        <p>Dummy page of Shell application</p>
      </main>
    </>
  );
}
```

**Reply by thmohd** [reply]
does this really work? how would that component show up as html head element?

**Reply by tmb-github** [reply]
It works. Try it. Next.js moves the title and meta elements to the bottom of the `<head>` upon navigation, replacing just those and leaving the other ones from the Root layout untouched.

**Reply by RichardeDev** [reply]
how to add  <scipt/> script balise

**Comment by calcgenie** [comment]
i have small website called https://www.splitle.com its about freelance recruitment but the problem is the technology that my company is using which wix. sometime im frustrated with wix. too much of <head> codes in every page without a purpose. will it be resolved in next js atleast. someone please give me headup on this. i have website called https://www.calcgenie.com but in this im facing different issue which is if i add google script for verification means, its showing some error. 

​https://www.splitle.com/

​
https://www.splitle.com/account-finance-industry-candidate

​
https://www.splitle.com/accounting-finance

​

https://www.splitle.com/accounting-finance-industry-recruiter

​

https://www.splitle.com/administrative-industry

​
https://www.splitle.com/administrative-industry-candidate

​
https://www.splitle.com/administrative-industry-recruiter

​
https://www.splitle.com/authorised-countries

​
https://www.splitle.com/blog
​

https://www.splitle.com/commissions

​
https://www.splitle.com/construction-industry

​
https://www.splitle.com/construction-industry-candidates
​

https://www.splitle.com/construction-industry-recruiter

 

https://www.splitle.com/contract-hire

 

https://www.splitle.com/contract-to-hire

​

https://www.splitle.com/direct-hire

​

https://www.splitle.com/engineering-industry


https://www.splitle.com/engineering-industry-candidate

https://www.splitle.com/engineering-industry-recruiter

https://www.splitle.com/faq

https://www.splitle.com/features


https://www.splitle.com/health-care-industry

https://www.splitle.com/health-care-industry-candidate

https://www.splitle.com/health-care-industry-recruiter


https://www.splitle.com/how-splitle-work

https://www.splitle.com/it-industry

https://www.splitle.com/it-industry-candidate

 
https://www.splitle.com/it-industry-recruiter

https://www.splitle.com/manufacturing-industry

​
https://www.splitle.com/manufacturing-industry-candidate

https://www.splitle.com/manufacturing-industry-recruiter

https://www.splitle.com/pharmaceutical-industry


https://www.splitle.com/pharmaceutical-industry-candidate

https://www.splitle.com/pharmaceutical-industry-recruiter

​
